### ISSUE TITLE
JavaScript error on current requests page (on collector server)

### ISSUE THREAD

**Issue by xpoinsard** [javamelody/javamelody#1237]
Trying to see current requests using the collector server 2.3.0 lead to an empty table and JavaScript error 

```
Uncaught TypeError: node is undefined
    getInnerText http://melody:8080/javamelody?resource=sorttable.js:211
    guessType http://melody:8080/javamelody?resource=sorttable.js:175
    makeSortable http://melody:8080/javamelody?resource=sorttable.js:89
    init http://melody:8080/javamelody?resource=sorttable.js:36
    forEach http://melody:8080/javamelody?resource=sorttable.js:459
    forEach http://melody:8080/javamelody?resource=sorttable.js:498
    init http://melody:8080/javamelody?resource=sorttable.js:34
javamelody:211:18
```

**Comment by evernat** [comment]
Do you have strack-trace of an exception in the server output of the collector server when displaying the current requests ?

**Comment by evernat** [comment]
not in storage directory, not in tomcat log files. The stack-trace of exception would be in the standard server output or in the error server output.

**Comment by xpoinsard** [comment]
Sorry, but no error in server output. I can see other errors, but none when requesting current requests.

**Comment by xpoinsard** [comment]
I started the collect server in debug and found the error : there is a null pointer exception because the threadName property of the the thread is null.
Here is the stack trace : 

> Daemon Thread [http-nio-8081-exec-15] (Suspended (exception NullPointerException))	
> 	owns: NioEndpoint$NioSocketWrapper  (id=2467)	
> 	I18N.htmlEncode(String, boolean, boolean) line: 137	
> 	I18N.htmlEncode(String, boolean) line: 155	
> 	HtmlAbstractReport.htmlEncode(String) line: 162	
> 	HtmlThreadInformationsReport.writeThreadWithStackTrace(String, List<StackTraceElement>) line: 208	
> 	HtmlCounterRequestContextReport.writeContext(CounterRequestContext, boolean) line: 302	
> 	HtmlCounterRequestContextReport.writeContexts(List<CounterRequestContext>) line: 281	
> 	HtmlCounterRequestContextReport.writeTitleAndDetails() line: 233	
> 	HtmlCoreReport.writeCurrentRequests(List<ThreadInformations>, List<CounterRequestContext>, boolean, int, boolean, Map<String,HtmlCounterReport>) line: 437	
> 	HtmlCoreReport.writeAllCurrentRequestsAsPart(Map<JavaInformations,List<CounterRequestContext>>) line: 422	
> 	HtmlReport.writeAllCurrentRequestsAsPart() line: 134	
> 	HtmlController.doAllCurrentRequestsAsPart() line: 161

**Comment by evernat** [comment]
@xpoinsard I think that your collector server 2.3.0 is monitoring an application using javamelody before 2.3.0 or before 1.99.1. Do you confirm that ?

**Comment by xpoinsard** [comment]
Yes the application is using javamleody 2.2.0

**Comment by evernat** [comment]
ok that's the cause. I will fix that.

**Comment by evernat** [comment]
fixed by ad512df
It's better than #1238 for pdf besides html
